third_party/move/move-model/src/ty.rs

```diff
@@ -519,7 +532,10 @@ impl Constraint {
             (Constraint::NoReference, Constraint::NoReference) => Ok(true),
             (Constraint::NoTuple, Constraint::NoTuple) => Ok(true),
             (Constraint::NoPhantom, Constraint::NoPhantom) => Ok(true),
-            (Constraint::HasAbilities(a1), Constraint::HasAbilities(a2)) => {
+            (
+                Constraint::HasAbilities(a1, params_exempted1),
```

Comment:
nit: maybe change the name `params_exempted` to `params_scope`?

Reply:
Done already